### DISCUSSION TITLE
Is it possible to create a fallback serializer?

### DISCUSSION THREAD

**Discussion by ojeytonwilliams** [fastify/fastify#5601] [Q&A]
Hi, we're using custom JSON schemas for each route, and I'd like to know if there's a smart way to serialize generic errors?

For example, we have a POST route with a response schema like this:

```
{
  response: {
    400: {
      type: 'object',
      properties: {
        error: {
          type: 'object',
          properties: {
            message: { type: 'string' },
            type: {
              anyOf: [
                {
                  const: 'AlreadyDonatingError',
                  type: 'string'
                },
                {
                  const: 'MethodRestrictionError',
                  type: 'string'
                }
              ]
            }
          },
          required: ['message', 'type']
        }
      },
      required: ['error']
    }
  }
};
```

but if the requester omits the body fastify tries to respond with `{"statusCode":400,"code":"FST_ERR_CTP_EMPTY_JSON_BODY","error":"Bad Request","message":"Body cannot be empty when content-type is set to 'application/json'"}`.

This isn't compatible with the schema and the response becomes a 500 `FST_ERR_FAILED_ERROR_SERIALIZATION`.

It's possible to work around this by modifying the schemas to accept the standard response, either by modifying each schema by hand (or using a helper function), or via [setSerializerCompiler](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/#serializer-compiler) (same thing, essentially).

I was wondering if this is a sensible solution or if there's an cleaner approach that I'm missing.

**Comment by metcoder95** [comment]
👋 

If looking to match all possible errors, including generic ones, you'll need to relax the strictness of your schemas, set a custom SerializerCompiler as you specify or provide a [custom error handler](https://fastify.dev/docs/latest/Reference/Server/#seterrorhandler) that will allow you to customize the error to your requirements before being serialized for a response.
